pkg/vpcmodel/output.go

```diff
@@ -89,37 +89,39 @@ func NewOutputGenerator(cConfigs *MultipleVPCConfigs, grouping bool, uc OutputUs
 			}
 			res.explanation = explanation
 		}
-		for i, vpcConfig := range cConfigs.Configs() {
-			if uc == AllEndpoints {
+		switch uc {
+		case AllEndpoints:
+			for i, vpcConfig := range cConfigs.Configs() {
 				nodesConn, err := vpcConfig.GetVPCNetworkConnectivity(grouping, res.lbAbstraction)
 				if err != nil {
 					return nil, err
 				}
 				res.nodesConn[i] = nodesConn
 			}
-			if uc == AllSubnets {
+		case AllSubnets:
+			for i, vpcConfig := range cConfigs.Configs() {
 				subnetsConn, err := vpcConfig.GetSubnetsConnectivity(true, grouping)
 				if err != nil {
 					return nil, err
 				}
 				res.subnetsConn[i] = subnetsConn
 			}
-			if uc == SubnetsDiff {
-				configsForDiff := &configsForDiff{vpcConfig, cConfigs.ConfigToCompare(i), Subnets}
-				configsDiff, err := configsForDiff.GetDiff()
-				if err != nil {
-					return nil, err
-				}
-				res.cfgsDiff = configsDiff
+		// diff: only comparison between single vpc configs is supported;
+		// thus instead of ranging over configs, takes the single config
+		case SubnetsDiff:
```

Comment:
Seems like the two diff use-cases can be united under `case SubnetsDiff. EndpointsDiff:`.
They only differ in one parameter in one function call.

Consider also merging the two report use-cases.

Reply:
I also prefer the non-merged version of the report use cases.

Reply:
report use case restored